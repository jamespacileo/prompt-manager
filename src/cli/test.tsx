import { Box, Text, render, useFocus } from "ink";

import React from "react";

//you need this
process.stdin.resume();

function Focus() {
  return (
    <Box flexDirection="column" padding={1}>
      <Box marginBottom={1}>
        <Text>
          Press Tab to focus next element, Shift+Tab to focus previous element,
          Esc to reset focus.
        </Text>
      </Box>
      <Item label="First" />
      <Item label="Second" />
      <Item label="Third" />
    </Box>
  );
}

function Item({ label }: any) {
  const { isFocused } = useFocus();
  return (
    <Text>
      {label} {isFocused && <Text color="green">(focused)</Text>}
    </Text>
  );
}

render(<Focus />);
