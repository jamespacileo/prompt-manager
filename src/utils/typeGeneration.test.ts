import { describe, expect, test } from "bun:test";
import type { JSONSchema7 } from "json-schema";
import type { IPrompt } from "../types/interfaces";
import {
	generateExportableSchemaAndType,
	generatePromptTypeScript,
	generatePromptTypescriptDefinition,
	generateTestInputs,
} from "./typeGeneration";

describe("Type Generation Utilities", () => {
	const sampleSchema: JSONSchema7 = {
		type: "object",
		properties: {
			name: { type: "string" },
			age: { type: "number" },
			isStudent: { type: "boolean" },
		},
		required: ["name", "age"],
	};

	test("generateExportableSchemaAndType", async () => {
		const result = await generateExportableSchemaAndType({
			schema: sampleSchema,
			name: "Person",
		});
		expect(result.formattedSchemaTs).toContain(
			"export const Person = z.object({",
		);
		expect(result.formattedSchemaTs).toContain("name: z.string(),");
		expect(result.formattedSchemaTs).toContain("age: z.number(),");
		expect(result.formattedSchemaTs).toContain(
			"isStudent: z.boolean().optional(),",
		);
		expect(result.formattedSchemaTs).toContain(
			"export type Person = z.infer<typeof Person>;",
		);
		expect(result).toMatchSnapshot();
	});

	test("generatePromptTypeScript", async () => {
		const prompt: IPrompt<any, any> = {
			name: "ExamplePrompt",
			category: "exampleCategory",
			description: "An example prompt",
			version: "1.0.0",
			template: "Example template",
			inputSchema: {
				type: "object",
				properties: {
					input: { type: "string" },
				},
			},
			outputSchema: {
				type: "object",
				properties: {
					output: { type: "string" },
				},
			},
			// parameters: {}, // Add appropriate parameters
			metadata: {
				created: "2023-01-01T00:00:00.000Z",
				lastModified: "2023-01-01T00:00:00.000Z",
			}, // Add appropriate metadata
			outputType: "structured", // Add appropriate outputType
			// configuration: {} // Add appropriate configuration
		};
		const result = await generatePromptTypeScript(prompt);
		expect(result).toContain("export interface ExamplePromptInput");
		expect(result).toContain("export interface ExamplePromptOutput");
		expect(result).toMatchSnapshot();
	});

	test("generatePromptTypescriptDefinition", async () => {
		const prompt: IPrompt<any, any> = {
			name: "ExamplePrompt",
			category: "exampleCategory",
			description: "An example prompt",
			version: "1.0.0",
			template: "Example template",
			inputSchema: {
				type: "object",
				properties: {
					input: { type: "string" },
				},
			},
			outputSchema: {
				type: "object",
				properties: {
					output: { type: "string" },
				},
			},
			metadata: {
				created: "2023-01-01T00:00:00.000Z",
				lastModified: "2023-01-01T00:00:00.000Z",
			},
			outputType: "structured",
		};
		const result = await generatePromptTypescriptDefinition(prompt);
		expect(result).toContain("export interface ExamplePromptInput");
		expect(result).toContain("export interface ExamplePromptOutput");
		expect(result).toMatchSnapshot();
	});

	test("generateTestInputs", () => {
		const result = generateTestInputs(sampleSchema, 3);
		expect(result).toHaveLength(3);
		result.forEach((input) => {
			expect(input).toHaveProperty("name");
			expect(input).toHaveProperty("age");
			expect(input).toHaveProperty("isStudent");
		});
		expect(result).toMatchSnapshot();
	});
});
