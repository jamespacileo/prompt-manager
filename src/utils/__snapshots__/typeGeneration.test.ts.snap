// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Type Generation Utilities generateExportableSchemaAndType 1`] = `
{
  "formattedSchemaTs": 
"

export const Person = z.object({
  name: z.string(),
  age: z.number(),
  isStudent: z.boolean().optional(),
});
export type Person = z.infer<typeof Person>;
"
,
  "formattedSchemaTsNoImports": 
"

export const Person = z.object({
  name: z.string(),
  age: z.number(),
  isStudent: z.boolean().optional(),
});
export type Person = z.infer<typeof Person>;
"
,
}
`;

exports[`Type Generation Utilities generatePromptTypeScript 1`] = `
"import {z} from "zod";
export interface ExamplePromptInput 

export const ExamplePromptInput = z.object({ input: z.string().optional() });
export type ExamplePromptInput = z.infer<typeof ExamplePromptInput>;


export interface ExamplePromptOutput 

export const ExamplePromptOutput = z.object({ output: z.string().optional() });
export type ExamplePromptOutput = z.infer<typeof ExamplePromptOutput>;

"
`;

exports[`Type Generation Utilities generateTestInputs 1`] = `
[
  {
    "age": 12754564.880375564,
    "isStudent": false,
    "name": "quis ea irure proident fugiat",
  },
  {
    "age": -45104176.67741235,
    "isStudent": false,
    "name": "voluptate ea proident",
  },
  {
    "age": 86756847.49900374,
    "isStudent": true,
    "name": "exercitation dolor irure officia",
  },
]
`;
